network/nxos/nxos_nxapi.py

```diff
@@ -0,0 +1,258 @@
+
+    if config.get('nxapi_status', 'Disabled') == 'Disabled':
+        if state == 'started':
+            # enable nxapi and get the new default config
+            commands.append('feature nxapi')
+            result['_config'] = dict()
+            result['changed'] = True
+            if module.params.get('check_mode'):
+                # make an assumption about default state
+                config['http_port'] = 80
+                config['sandbox_status'] = 'Disabled'
+            else:
+                # get the default config
+                execute_commands(module, commands)
+                config = module.from_json(module.execute(['show nxapi | json'])[0])
+        else:
+            # nxapi already disabled
+            return result
+    elif config.get('nxapi_status', 'Disabled') == 'Enabled' and state == 'stopped':
+        # disable nxapi and exit
+        commands.append('no feature nxapi')
+        result['changed'] = True
+        result['commands'] = commands
+        execute_commands(module, commands)
+        return result
+
+    # configure http and https
+    for protocol in ['http', 'https']:
+        cmds, chg = http_commands(protocol, module.params['{0}_port'.format(protocol)],
+                                  module.params[protocol], config)
+        if chg:
+            commands += cmds
+            result['changed'] = True
+
+    # configure sandbox
+    config_sandbox = config.get('sandbox_status', None)
+    enable_sandbox = module.params.get('sandbox')
+
+    if config_sandbox is None:
+        # there is no prior state, so we must set one
+        result['changed'] = True
+        if enable_sandbox:
+            commands.append('nxapi sandbox')
+        else:
+            commands.append('no nxapi sandbox')
+    else:
+        # there is a prior state, so be idempotent
+        if config_sandbox == 'Enabled' and not enable_sandbox:
+            # turn off sandbox
+            commands.append('no nxapi sandbox')
+            result['changed'] = True
+        elif config_sandbox == 'Disabled' and enable_sandbox:
+            # turn on sandbox
+            commands.append('nxapi sandbox')
+            result['changed'] = True
+
+    if len(commands) > 0:
+        # something requires change
+        result['commands'] = commands
+        execute_commands(module, commands)
+
+    return result
+
+
+def main():
+    """ main entry point for module execution
+    """
+
+    argument_spec = dict(
+        state=dict(default='started', choices=['started','stopped']),
+        http_port=dict(default=80, type='int'),
+        https_port=dict(default=443, type='int'),
+        http=dict(aliases=['enable_http'], default=True, type='bool'),
+        https=dict(aliases=['enable_https'], default=True, type='bool'),
+        sandbox=dict(aliases=['enable_sandbox'], default=True, type='bool'),
+
+        # Only allow configuration of NXAPI using cli transpsort
+        transport=dict(required=True, choices=['cli'])
+    )
```

Comment:
Overriding transport param raises the exception before nxos.py gets evaluated. If nxos.py is evaluated by the interpreter, then we need urls.py from module_utils included so that fetch_url is available. And even with that there is still some other exception getting raised that I couldn't track down. But it's an exception that makes no sense to the end user, so we would have to include something else or fix whatever it is. Might be including net_cfg.py.

Anyway, this throws an exception letting user know that the only transport choice is 'cli', which seems reasonable enough.